fw-cfg の signature "VMMV" に対応する。
fw-cfg の index 0x1 4byte interface version がなくても動作するようにする。

Index: uefi/OvmfPkg/Library/QemuFwCfgLib/QemuFwCfgPeiDxe.c
===================================================================
--- uefi.orig/OvmfPkg/Library/QemuFwCfgLib/QemuFwCfgPeiDxe.c	2014-01-04 04:57:26.239662000 +0900
+++ uefi/OvmfPkg/Library/QemuFwCfgLib/QemuFwCfgPeiDxe.c	2015-01-06 11:22:08.725736448 +0900
@@ -57,12 +57,17 @@
   QemuFwCfgSelectItem (QemuFwCfgItemSignature);
   Signature = QemuFwCfgRead32 ();
   DEBUG ((EFI_D_INFO, "FW CFG Signature: 0x%x\n", Signature));
+
+  if (Signature == SIGNATURE_32 ('V', 'M', 'M', 'V')) {
+    DEBUG ((EFI_D_INFO, "FwCfg interface is supported.\n"));
+    return RETURN_SUCCESS;
+  }
+
   QemuFwCfgSelectItem (QemuFwCfgItemInterfaceVersion);
   Revision = QemuFwCfgRead32 ();
   DEBUG ((EFI_D_INFO, "FW CFG Revision: 0x%x\n", Revision));
   if ((Signature != SIGNATURE_32 ('Q', 'E', 'M', 'U')) ||
-      (Revision < 1)
-     ) {
+      (Revision < 1)) {
     DEBUG ((EFI_D_INFO, "QemuFwCfg interface not supported.\n"));
     mQemuFwCfgSupported = FALSE;
     return RETURN_SUCCESS;
Index: uefi/OvmfPkg/Library/QemuFwCfgLib/QemuFwCfgSec.c
===================================================================
--- uefi.orig/OvmfPkg/Library/QemuFwCfgLib/QemuFwCfgSec.c	2014-01-04 04:57:36.727208000 +0900
+++ uefi/OvmfPkg/Library/QemuFwCfgLib/QemuFwCfgSec.c	2015-01-06 11:23:18.791611266 +0900
@@ -42,6 +42,12 @@
   QemuFwCfgSelectItem (QemuFwCfgItemSignature);
   Signature = QemuFwCfgRead32 ();
   DEBUG ((EFI_D_INFO, "FW CFG Signature: 0x%x\n", Signature));
+
+  if (Signature == SIGNATURE_32 ('V', 'M', 'M', 'V')) {
+    DEBUG ((EFI_D_INFO, "FwCfg interface is supported.\n"));
+    return TRUE;
+  }
+
   QemuFwCfgSelectItem (QemuFwCfgItemInterfaceVersion);
   Revision = QemuFwCfgRead32 ();
   DEBUG ((EFI_D_INFO, "FW CFG Revision: 0x%x\n", Revision));
